contents/csharp/CSAsyncAwait.md

```diff
@@ -0,0 +1,322 @@
+<frontmatter>
+  title: Asynchronous Programming In C# Using Async/Await
+  header: pagetop.md
+  footer: footer.md
+  head: head.md
+  siteNav: mainNav.md
+  pageNav: 3
+</frontmatter>
+
+<div class="website-content">
+
+{{ booktitle | safe }}
+
+# Asynchronous Programming In C# Using Async/Await
+
+**Author: [Muhammad Irham Rasyidi Bin Zainal](https://github.com/xantho09)**<br>
+
+<box id="article-toc">
+
+* [What Is Asynchronous Programming](#what-is-asynchronous-programming)
+* [Example Of Asynchronous Programming](#example-of-asynchronous-programming)
+* [Basic Syntax‎](#basic-syntax)
+    * [Async](#async)
+    * [Await](#await)
+    * [Returning Values](#returning-values)
+* [Control Flow‎](#control-flow)
+* [Disadvantages Of Async/Await](#disadvantages-of-asyncawait)
+    * [Potential Pitfall: Deadlock](#potential-pitfall-deadlock)
+    * [Additional Overhead](#additional-overhead)
+* [Resources](#resources)
+* [Further Reading](#further-reading)
+
+</box>
+
+## What Is Asynchronous Programming?
+
+In **synchronous programming**, code is executed one step at a time, waiting for each step to be finished before proceeding to the next step. For example, if the code looks something like this:

```

Comment:
Might want to clarify whether the steps are dependent on each other/need to be return their results in a specified order for this example.

Might also want to clarify that async and result in steps finishing in a different order.